packages/neon_framework/packages/account_repository/lib/src/account_repository.dart

```diff
@@ -0,0 +1,329 @@
+
+        return (active: e.active, accounts: accounts);
+      }).asBroadcastStream();
+
+  /// Whether accounts are logged in.
+  bool get hasAccounts {
+    return _accounts.hasValue && _accounts.value.accounts.isNotEmpty;
+  }
+
+  /// Retrieves the account associated with the given [accountID].
+  Account? accountByID(String accountID) {
+    return _accounts.value.accounts[accountID];
+  }
+
+  /// Checks whether the provided [url] is a valid login qr code.
+  static bool isLogInQRCode({required String url}) {
+    return LoginQRcode.tryParse(url) != null;
+  }
+
+  /// Loads the stored account information.
+  ///
+  /// If specified [rememberLastUsedAccount] takes precedence over [initialAccount].
+  /// If the stored or provided initial account are not in the list of stored credentials
+  /// they are silently ignored and the default of `first account in the list` is used.
+  Future<void> loadAccounts({
+    String? initialAccount,
+    bool rememberLastUsedAccount = false,
+  }) async {
+    final stored = await _storage.readAccounts();
+    final accounts = BuiltMap<String, Account>.build((b) {
+      for (final credentials in stored) {
+        b[credentials.id] = Account((b) {
+          b
+            ..credentials.replace(credentials)
+            ..client = buildClient(
+              httpClient: _httpClient,
+              userAgent: _userAgent,
+              credentials: credentials,
+            );
+        });
+      }
+    });
+
+    var active = initialAccount;
+    if (rememberLastUsedAccount) {
+      active = await _storage.readLastAccount();
+    }
+
+    if (active == null || !accounts.containsKey(active)) {
+      active = accounts.keys.firstOrNull;
+    }
+
+    _accounts.add(
+      (
+        active: active,
+        accounts: accounts,
+      ),
+    );
+  }
+
+  // TODO: this doesn't really belong here.
+  /// Fetches the status of the server.
+  ///
+  /// May throw a [FetchStatusFailure].
+  Future<core.Status> getServerStatus(Uri serverURL) async {
+    final client = buildUnAuthenticatedClient(
+      httpClient: _httpClient,
+      userAgent: _userAgent,
+      serverURL: serverURL,
+    );
+
+    try {
+      final response = await client.authentication.core.getStatus();
+      return response.body;
+    } on http.ClientException catch (error, stackTrace) {
+      Error.throwWithStackTrace(FetchStatusFailure(error), stackTrace);
+    }
+  }
+
+  /// Initializes the Nextcloud login flow.
+  ///
+  /// May throw a [InitLoginFailure].
+  Future<Uri> loginFlowInit(Uri serverURL) async {
+    final client = buildUnAuthenticatedClient(
+      httpClient: _httpClient,
+      userAgent: _userAgent,
+      serverURL: serverURL,
+    );
+
+    try {
+      final initResponse = await client.authentication.clientFlowLoginV2.init();
+
+      final pollBody = core.ClientFlowLoginV2PollRequestApplicationJson((b) {
+        b.token = initResponse.body.poll.token;
+      });
+      _flow = (client, pollBody);
```

Comment:
My main concern is that we're going to build a new client for every poll make (currently once a second).
Creating a client isn't as expensive as it was beccause we now reuse the base client but it still requires creating:
- new NeonHttpClient
    - new BaseHeaderInterceptor
    - new CSRFInterceptor
- new NextcloudClient
    - new nextcloud core client

Sending the request dierectly isn't a good option as it wouldn't set the useragent or timeout.

Reply:
> Sending the request dierectly isn't a good option as it wouldn't set the useragent or timeout.

Not if the NeonHttpClient is used

Can we instead create the client for this outside the repository, so both methods just get it passed and we only need a single instance that isn't stored in the repository?

Reply:
And how about returning the flow instead of storing it in the repository at all? Having this hidden state feels weird to me, it should be more explicit by returning and passing it again.

Reply:
I've changed it for now (build a new client and return the token).
This isn't any hotpath in the app, and we don't expect users to only use the login flow :)

We can still change it later on.